lib/adapters/http.js

```diff
@@ -185,7 +185,7 @@ module.exports = function httpAdapter(config) {
         stream.on('end', function handleStreamEnd() {
           var responseData = Buffer.concat(responseBuffer);
           if (config.responseType !== 'arraybuffer') {
-            responseData = responseData.toString('utf8');
+            responseData = responseData.toString(config.charset || 'utf-8');
```

Comment:
The default is `utf8` (without the dash) but as it's the Node Buffer default value you don't need to set the `|| 'utf8`, you can just pass the `config.charset`.